### DISCUSSION TITLE
First boot seems to be unsuccessfull using a custom pcb

### DISCUSSION THREAD

**Discussion by 666dsa666** [darthcloud/BlueRetro#170] [Q&A]
Hello,

I'm sorry to post about my issue here as it might not be a project issue but maybe the PCB I'm using.

https://oshpark.com/shared_projects/aW1v2Ei3

I've done 5 of them today and am having issue with all of them not working.

I tried my best and was successful using a spare ESP32 without the board...

When using the board I was able to get those logs from putty : 
`Brownout detector was triggered

▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒                                                                                                                                                          ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒                                                                                                                                                          ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒                                                                                                                                                          ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒                                                                                                                                                          ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒                                                                                                                                                          ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒I (237) cpu_start: Pro cpu up.
I (237) cpu_start: Single core mode
I (245) cpu_start: Pro cpu start user code
I (245) cpu_start: cpu freq: 240000000
I (245) cpu_start: Application information:
I (245) cpu_start: Project name:     BlueRetro
I (246) cpu_start: App version:      v1.2.2 psx_ps2_spiffs
I (247) cpu_start: Compile time:     Jan 19 2022 22:50:38
I (248) cpu_start: ELF file SHA256:  61344aea9b5c82db...
I (248) cpu_start: ESP-IDF:          v4.4-dev-2128-gf45b60b28b
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒I (237) cpu_start: Pro cpu up.
I (237) cpu_start: Single core mode
I (245) cpu_start: Pro cpu start user code
I (245) cpu_start: cpu freq: 240000000
I (245) cpu_start: Application information:
I (245) cpu_start: Project name:     BlueRetro
I (246) cpu_start: App version:      v1.2.2 psx_ps2_spiffs
I (247) cpu_start: Compile time:     Jan 19 2022 22:50:38
I (248) cpu_start: ELF file SHA256:  61344aea9b5c82db...
I (248) cpu_start: ESP-IDF:          v4.4-dev-2128-gf45b60b28b
I (249) cpu_start: App cpu up.
I (250) heap_init: Initializing. RAM available for dynamic allocation:
I (251) heap_init: At 3FF80000 len 00002000 (8 KiB): RTCRAM
I (251) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (252) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (253) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (254) heap_init: At 3FFD0198 len 0000FE68 (63 KiB): DRAM
I (254) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (255) heap_init: At 3FFE4350 len 00003A48 (14 KiB): D/IRAM
I (256) heap_init: At 3FFE7E28 len 000181D8 (96 KiB): D/IRAM
I (257) heap_init: At 40078000 len 00008000 (32 KiB): IRAM
I (258) heap_init: At 4009D774 len 0000288C (10 KiB): IRAM
I (259) spi_flash: detected chip: generic
I (260) spi_flash: flash io: dio
I (260) esp_core_dump_uart: Init core dump to UART
I (260) esp_core_dump_uart: Init core dump to UART
I (261) cpu_start: Starting scheduler on PRO CPU.
App main at 40082FFC
# bt_host_load_bdaddr_from_file: No BDADDR on SD. Using ESP32's MAC
I (323) BTDM_INIT: BT controller compile version [d874f68]
I (323) system_api: Base MAC address is not set
I (323) system_api: read default base MAC address from EFUSE
I (323) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07
W (323) phy_init: failed to load RF calibration data (0x1102), falling back to full calibration
`

I haven't tried using the kit on my console yet and wonder if the issue is not because of the power supply and added parts that might cause issue with it.

I'll probably be able to test it next week but hopefully someone will have an answer about my issue before.

Thank you !

**Comment by darthcloud** [comment]
Moved this to discussion as its isn't a FW bug.

@666dsa666 Can you post a picture of how you are powering the board?

**Reply by 666dsa666** [reply]
I power it using USB for testing purpose

I saw a message earlier about a via on 3.3v, the 3,3v and int pads are actually not present
I power them via USB : 
![PXL_20220217_090649348](https://user-images.githubusercontent.com/11072018/154443539-aca86e73-dfea-4f63-a9d1-4bb1c125d5b8.jpg)
 
By the way I tried troubleshooting things a bit : 

- Issue seems to be only on first boot
- If I am somehow able to get it to boot once and configure it using the web configuraton issue does not happen anymore
- If I do not use the PCB on first boot there is no issue (sadly I soldered 4 of them already)

**Comment by 666dsa666** [comment]
Trying to power it using my PS1 doesn't seems to work.

Adapter simply doesn't turn on, I'll try on a ps2 today.

**Reply by 666dsa666** [reply]
Ok so PS2 is a no go either, I'll try another controller plug

**Reply by 666dsa666** [reply]
Sadly I can't seems to get anything to work.

I'm doing my best to troubleshoot it but I am pretty sure I am missing something...

8v from my PS1 ans both my PS2 (slim and fat) only output 0.8v which is definitely not enough for the LDO to power the controller.

I tested the LDO with a 5v input and do get a 3.5v output which is what the documentation states but I can't believe all three of my consoles have an issue.

**Comment by darthcloud** [comment]
I'm not familiar with this HW so hard to help.

If you read 0V on 8V pin with adaptor on 3 different console, you must have a short on your board somewhere.

Those LDO come in a multiple variant with very similar part number but completely different pinout.

Make sure you got the exact part number and not the "reverse" aka "R" variant, or the other way around if this board need the reverse one.

People had issue powering from 8V on the #102 PU-18 rev PSX. So maybe try removing the LDO completely and use console 3.3v on the LDO 3.3v pad.

**Reply by darthcloud** [reply]
maybe @Micha213 can help?

**Reply by Micha213** [reply]
I had a similar problem with some ESP32 d1 mini boards. I was able to flash them but BlueRetro wouldn't boot. Sometimes the boot works too, but only very rarely. The brown out detection was triggered. The ESP32 needs too much power to start up and the onboard 3.3V LDO on the d1 mini board can't handle it. So the problem is with the 3.3V LDO on the d1 mini board.

The original plan was to convert the 8V of the PS1/2 to 5V with the LDO on the shield. This 5V then goes into the 5V pin of the d1 mini board and is converted there by the 3.3V LDO. So from 8V to 5V and then from 5V to 3.3V.

To work around the problem with the 3.3V LDO on the d1 mini board, I swapped the 5V LDO on the shield for a 3.3V LDO and connected the output of the LDO directly to the 3.3V pin of the ESP32. So it is converted directly from 8V to 3.3V on the shield. This is how I was able to fix the brown out detection problem.

For this I had also created a new version 1.2 of the Shild.
[link](https://oshpark.com/shared_projects/aharS64j)
More information:
[link](https://github.com/darthcloud/BlueRetro/discussions/26#discussioncomment-972352)
There you can use jumper pads to choose whether you want to install a 5V or 3.3V LDO on the shield.
But you can also retrofit it with the v1.1 Shield.
To do this you have to swap the 5V LDO on the shield for a 3.3V LDO. The trace from the output of the LDO to the 5V pin must be interrupted. And a new connection from the LDO output to the 3.3V pin of the d1 mini.

However, all this doesn't explain why you only get 0.8V on the 8V pin of the PS1/2. Does this only happen if you connect the ESP32 to the PS1/2 or also without it?

I hope I could help
![154443539-aca86e73-dfea-4f63-a9d1-4bb1c125d5b8](https://user-images.githubusercontent.com/46410328/154526090-1b1bc24f-6d31-41eb-992f-dfa48e667f3b.jpg)

**Reply by 666dsa666** [reply]
Thank you for your answer, and thank you for your pcb !

I was able to make a few tests using a USB Tester but couldn't understand what was happening but your answer gives me a bit more informations.

When booting correctly I get a reading of ~5v 120mA
When not booting : ~5v 80mA
When booting incorrectly : ~5v 130mA (with some fluctuations when the ESP might be rebooting)

I ordered a few 3.3v LDO, hopefully this will solve my issue but they will not be there before a few weeks at least.

I will be checking my consoles output and get back to you in the meantime.

**Reply by Micha213** [reply]
Exactly, I took the same measurements. If BlueRetro is booted normally, the ESP32 consumes about 120mA. When the brownout detection triggered, it was only 80mA. However, the ESP32 needs at least 350-400mA to boot. The small 3.3V LDO from the d1 mini board cannot do that.

**Reply by 666dsa666** [reply]
Update, the PS1 8v fuse is dead, I'll check PS2 fuses soon.
